File: problemas/ad-hoc/1136/1136.py
while True:
    try:
        N, B = [int(x) for x in input().strip().split(' ')]
        bolas = [int(x) for x in input().strip().split(' ')]

        possibilidades = set()
        possibilidades.add(0)

        for i in range(B):
            for j in range(i + 1, B):
                possibilidades.add(abs(bolas[i] - bolas[j]))
        
        print('Y' if len(possibilidades) == N + 1 else 'N')
    except EOFError:
        break

File: problemas/ad-hoc/1136/1136.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.trim().split('\n');

lines.pop();

while(lines.length){
    let [N, B] = lines.shift().trim().split(' ').map((x) => parseInt(x));
    let bolas = lines.shift().trim().split(' ').map((x) => parseInt(x));

    let possibilidades = new Set();
    possibilidades.add(0);

    for(let i = 0; i < B; ++i){
        for(let j = i + 1; j < B; ++j){
            possibilidades.add(Math.abs(bolas[i] - bolas[j]));
        }
    }

    console.log(possibilidades.size === N + 1 ? 'Y' : 'N');
}
